In [1]:
import pandas as pd 
import numpy as np

## Load the Dataset

In [3]:
train_data = pd.read_csv("train.csv")
train_data.sample(5)

PassengerId  Survived  Pclass                                      Name  \
296          297         0       3                        Hanna, Mr. Mansour   
727          728         1       3                  Mannion, Miss. Margareth   
699          700         0       3  Humblen, Mr. Adolf Mathias Nicolai Olsen   
243          244         0       3             Maenpaa, Mr. Matti Alexanteri   
634          635         0       3                        Skoog, Miss. Mabel   

        Sex   Age  SibSp  Parch             Ticket     Fare  Cabin Embarked  
296    male  23.5      0      0               2693   7.2292    NaN        C  
727  female   NaN      0      0              36866   7.7375    NaN        Q  
699    male  42.0      0      0             348121   7.6500  F G63        S  
243    male  22.0      0      0  STON/O 2. 3101275   7.1250    NaN        S  
634  female   9.0      3      2             347088  27.9000    NaN        S

In [5]:
# test data
test_data = pd.read_csv("test.csv")
test_data.sample(5)

PassengerId  Pclass                                    Name     Sex  \
264         1156       2  Portaluppi, Mr. Emilio Ilario Giuseppe    male   
121         1013       3                       Kiernan, Mr. John    male   
380         1272       3                   O'Connor, Mr. Patrick    male   
266         1158       1   Chisholm, Mr. Roderick Robert Crispin    male   
186         1078       2    Phillips, Miss. Alice Frances Louisa  female   

      Age  SibSp  Parch       Ticket     Fare Cabin Embarked  
264  30.0      0      0   C.A. 34644  12.7375   NaN        C  
121   NaN      1      0       367227   7.7500   NaN        Q  
380   NaN      0      0       366713   7.7500   NaN        Q  
266   NaN      0      0       112051   0.0000   NaN        S  
186  21.0      0      1  S.O./P.P. 2  21.0000   NaN        S

In [12]:
train_data.describe

<bound method NDFrame.describe of      PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                 

In [13]:
test_data.describe

<bound method NDFrame.describe of      PassengerId  Pclass                                          Name  \
0            892       3                              Kelly, Mr. James   
1            893       3              Wilkes, Mrs. James (Ellen Needs)   
2            894       2                     Myles, Mr. Thomas Francis   
3            895       3                              Wirz, Mr. Albert   
4            896       3  Hirvonen, Mrs. Alexander (Helga E Lindqvist)   
..           ...     ...                                           ...   
413         1305       3                            Spector, Mr. Woolf   
414         1306       1                  Oliva y Ocana, Dona. Fermina   
415         1307       3                  Saether, Mr. Simon Sivertsen   
416         1308       3                           Ware, Mr. Frederick   
417         1309       3                      Peter, Master. Michael J   

        Sex   Age  SibSp  Parch              Ticket      Fare Cabin Embarked 

Now, We have all **891 rows and 12 columns** in train.csv & **418 rows and 11 columns** in test.csv

## Our First Submission

Our Goal is to find patterns in **train.csv** that help us predict whether the passengers in **test.csv** survived.

It might initially feel overwhelming to look for patterns, when there's so much data sort through. So, we'll start simple.

## Let's Find a Pattern
Keep in mind that the sample submission file, **gender_submission.csv**, operates under the assumption that every female passenger survived while all male passengers did not.

In this a sensible initial assumption? We will verify if this trend is consistent with the data found in train.csv